### PR TITLE
📝(doc) update README

### PR DESCRIPTION
**Pull request by madmatah**
## Purpose

We would like to have a documentation explaining how to initialize the
project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,146 @@
...
+# Jitsi-k8s, a scalable Jitsi deployment for Kubernetes
 
-The goal of this project is to deploy a scalable Jitsi infrastructure with
-Terraform and Kubernetes.
+## Overview
+
+The goal of this project is to deploy a scalable [Jitsi](https://github.com/jitsi/jitsi-meet)
+infrastructure on a Kubernetes cluster.
+
+In its current state, this projects allow to deploy the "Jibri" component.
```
Comment (sampaccoud) [inline README.md:8]:
```suggestion
In its current state, this project only allows to deploy the "Jibri" component.
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,146 @@
...
+
+## Architecture
+
+This repository is composed of two parts :
```
Comment (sampaccoud) [inline README.md:12]:
```suggestion
This repository is composed of two parts:
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,146 @@
...
+
+ - A [terraform](https://www.terraform.io/) project, whose goal is to deploy a ready to use Kubernetes
```
Comment (sampaccoud) [inline README.md:14]:
```suggestion
 - A [terraform](https://www.terraform.io/) project, whose goal is to deploy a ready-to-use Kubernetes
```

File: README.md [hunk 4]
```diff
@@ -1,4 +1,146 @@
...
+   cluster. It also installs some Helm packages to enable required features (e.g. custom metrics)
+
+ - A [kustomize](https://kustomize.io/) project, in which are defined the resources to deploy on
+   the Kubernetes cluster.
+
+
+We made the following (opinionated) choices, but the project can be easily adapted to chose alternatives options :
```
Comment (sampaccoud) [inline README.md:21]:
```suggestion
We made the following (opinionated) choices, but the project can be easily adapted to choose alternative options :
```

File: README.md [hunk 5]
```diff
@@ -1,4 +1,146 @@
...
+
+- We use [Scaleway](https://www.scaleway.com/) as a cloud provider, to provision the Kubernetes cluster.
+- The terraform state is stored on an [Openstack Swift](https://docs.openstack.org/swift/latest/) bucket.
```
Comment (sampaccoud) [inline README.md:24]:
```suggestion
- The terraform state is stored in an [Openstack Swift](https://docs.openstack.org/swift/latest/) bucket.
```

File: README.md [hunk 6]
```diff
@@ -1,4 +1,146 @@
...
+
+
+## Getting started
+
+### Terraform configuration
+
+#### 1.1 Configure Openstack credentials
+
+The terraform state is stored on an Openstack Swift bucket.
```
Comment (sampaccoud) [inline README.md:33]:
```suggestion
The Terraform state is stored in an Openstack Swift bucket.
```

File: README.md [hunk 7]
```diff
@@ -1,4 +1,146 @@
...
+
+Execute the following command to initialize the `env.d/terraform` file:
+
+```
+make env.d/terraform
+```
+
+Then, edit this file (`env.d/terraform`) and define the `OS_*` variables according to your Openstack environment.
+
+Note: if you use OVH as Swift bucket provider, you can get these values in the Horizon interface, by clicking on the «Openstack RC File v3» on the upper-right menu.
```
Comment (sampaccoud) [inline README.md:44]:
```suggestion
Note: if you use OVH as Swift bucket provider, you can get these values in the Horizon interface, by clicking on the «Openstack RC File v3» in the upper-right menu.
```

File: README.md [hunk 8]
```diff
@@ -1,4 +1,146 @@
...
+
+#### 1.2 Create the terraform state bucket
```
Comment (sampaccoud) [inline README.md:46]:
```suggestion
#### 1.2 Create the Terraform state bucket
```

File: README.md [hunk 9]
```diff
@@ -1,4 +1,146 @@
...
+
+Important: if your team already have initialized the state bucket, skip this part and go to chapter `1.3`
```
Comment (sampaccoud) [inline README.md:48]:
```suggestion
Important: if your team has already initialized the state bucket, skip this part and go to chapter `1.3`
```

File: README.md [hunk 10]
```diff
@@ -1,4 +1,146 @@
...
+
+To create the Swift bucket required to store the terraform state, there is a
+terraform project dedicated to this in the directory `terraform/create_state_bucket`.
```
Comment (sampaccoud) [inline README.md:51]:
```suggestion
To create the Swift bucket required to store the Terraform state, there is a
Terraform project dedicated to this in the directory `terraform/create_state_bucket`.
```

File: README.md [hunk 11]
```diff
@@ -1,4 +1,146 @@
...
+
+To create the state bucket, you just have to execute the following commands:
```
Comment (sampaccoud) [inline README.md:53]:
```suggestion
To create the state bucket, you need to execute the following commands:
```

File: README.md [hunk 12]
```diff
@@ -1,4 +1,146 @@
...
+
+```bin/state init```
+
+and then:
+
+```bin/state apply```
+
+#### 1.3 Initialize terraform
```
Comment (sampaccoud) [inline README.md:61]:
```suggestion
#### 1.3 Initialize Terraform
```

File: README.md [hunk 13]
```diff
@@ -1,4 +1,146 @@
...
+
+Execute the following command to initialize terraform with the configured state.
```
Comment (sampaccoud) [inline README.md:63]:
```suggestion
Execute the following command to initialize Terraform with the configured state.
```

File: README.md [hunk 14]
```diff
@@ -1,4 +1,146 @@
...
+
+```bin/terraform init```
+
+#### 1.4. Scaleway credentials
+
+We use Scaleway to provision the Kubernetes cluster. You need to set your Scaleway API credentials in environment variables.
+
+##### 1.4a : single environment
+
+If you intend to deploy this project on a single environment, you can set the following variables  in `env.d/terraform` :
```
Comment (sampaccoud) [inline README.md:73]:
```suggestion
If you intend to deploy this project on a single environment, you can set the following variables in `env.d/terraform` :
```

File: README.md [hunk 15]
```diff
@@ -1,4 +1,146 @@
...
+
+- `SCW_DEFAULT_PROJECT_ID` : your scaleway project ID
+- `SCW_ACCESS_KEY` : your Scaleway API access key
+- `SCW_SECRET_KEY` : your Scaleway API secret key
+
+##### 1.4b : multiple environment (recommended)
```
Comment (sampaccoud) [inline README.md:79]:
```suggestion
##### 1.4b : multiple environments (recommended)
```

File: README.md [hunk 16]
```diff
@@ -1,4 +1,146 @@
...
+
+In most cases, you'll deploy this project on multiple environments (e.g. : preprod, production).
+
+You'll have to create a Scaleway project for each of these environment.
+And each environment will have its own terraform workspace.
```
Comment (sampaccoud) [inline README.md:84]:
```suggestion
You'll have to create a Scaleway project for each of these environments.
And each environment will have its own Terraform workspace.
```

File: README.md [hunk 17]
```diff
@@ -1,4 +1,146 @@
...
+```
+TF_VAR_scaleway_access_key={ production = "access_key_A", preprod = "access_key_B"}
+TF_VAR_scaleway_secret_key={ production = "secret_key_A", preprod = "secret_key_B"}
+TF_VAR_scaleway_project_id={ production = "project_A", preprod = "project_B"}
+```
+
+#### 1.5 : customize configuration
+
+Look at the variables defined in `terraform/variables.tf`.
+You can customize them per environment, just like we did in step `1.4b`.
```
Comment (sampaccoud) [inline README.md:109]:
```suggestion
You can customize them by environment, just like we did in step `1.4b`.
```

File: README.md [hunk 18]
```diff
@@ -1,4 +1,146 @@
...
+```bin/terraform plan```
+
+And when you are ready to deploy it:
+
+```bin/terraform apply```
+
+
+### Kubernetes
+
+Once your kubernetes cluster is deployed, you can interract with it with the command `bin/kubectl`.
```
Comment (sampaccoud) [inline README.md:124]:
```suggestion
Once your kubernetes cluster is deployed, you can interact with it with the command `bin/kubectl`.
```

**Review by sampaccoud — APPROVED** [review]

**Comment by madmatah** [comment]
@sampaccoud Thanks for your review!

**Pull request merged**